src/webapp/pages/current-call/CurrentCallPage.tsx

```diff
@@ -1,47 +1,77 @@
 import { CurrentCallContent } from "../../components/current-call/CurrentCallContent";
+import { useSpecificCall } from "../../hooks/useSpecificCall";
+import { statusMap } from "../../components/current-call/StatusMap";
 
-interface CurrentCallPageContentProps {
+interface CurrentCallPageProps {
     moduleName: string;
 }
 
-export const CurrentCallPage: React.FC<CurrentCallPageContentProps> = React.memo(({ moduleName }) => {
-    return (
-        <MainLayout>
-            <CurrentCallPageContent moduleName={moduleName} />
-        </MainLayout>
-    );
-});
+interface CurrentCallPageContentProps {
+    moduleId: string;
+    moduleName: string;
+}
 
-export const CurrentCallPageContent: React.FC<CurrentCallPageContentProps> = React.memo(({ moduleName }) => {
+export const CurrentCallPage: React.FC<CurrentCallPageProps> = React.memo(({ moduleName }) => {
     const { compositionRoot } = useAppContext();
 
-    // TODO: replace useGlassModule (or parameters) with actual hook to fetch current call data
     const result = useGlassModule(compositionRoot, moduleName);
+    switch (result.kind) {
+        case "loading":
+            return <CircularProgress />;
+        case "error":
+            return <Typography variant="h6">{result.message}</Typography>;
+        case "loaded":
+            return (
+                <MainLayout>
+                    <CurrentCallPageContent moduleName={moduleName} moduleId={result.data.id} />
+                </MainLayout>
+            );
+    }
+});
+
+export const CurrentCallPageContent: React.FC<CurrentCallPageContentProps> = React.memo(({ moduleId, moduleName }) => {
+    const location = useLocation();
+    const queryParameters = new URLSearchParams(location.search);
+
+    //TO DO : orgUnit and period will not be fetched from queryParameters, it will be fetched from context.
+    const periodVal = queryParameters?.get("period");
+    const orgUnitVal = queryParameters.get("orgUnit");
+
+    //set default values till the context changes are integrated
+    const [period, setPeriod] = useState(periodVal === null ? 2023 : parseInt(periodVal));
+    const [orgUnit, setOrgUnit] = useState(orgUnitVal === null ? "DVnpk4xiXGJ" : orgUnitVal);
+
+    const { compositionRoot } = useAppContext();
+    const currentCall = useSpecificCall(compositionRoot, moduleId, orgUnit, period);
 
     const handleClick = (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
         event.preventDefault();
+        setPeriod(0); //new period value
+        setOrgUnit("new orgUnit value");
     };
 
-    switch (result.kind) {
+    switch (currentCall.kind) {
         case "loading":
             return <CircularProgress />;
         case "error":
-            return <Typography variant="h6">{result.message}</Typography>;
-        case "loaded":
+            return <Typography variant="h6">{currentCall.message}</Typography>;
+        case "loaded": {
+            const currentCallDetails = statusMap.get(currentCall.data);
```

Comment:
The logic to convert from status to status details seems presentation logic. The custom hooks are the presentation adapters to convert the data from the domain to the presentation.

The components only should render without applying any logic.

The custom hook can be renamed from useSpecificCall to useStatusCall

Reply:
Moved the presentation logic to custom hook and renamed it.